### ISSUE TITLE
serde_derive deserialize_with untagged enum regression

### ISSUE THREAD

**Issue by arcnmx** [serde-rs/serde#1268]
I still need to create a simple reproduction for the issue, and not sure if this is covered by another issue report, but effectively...

```rust
#[derive(Deserialize)]
#[serde(untagged)]
enum E {
    Variant(
        #[serde(deserialize_with="...")]
        Ty
    ),
}
```

Now fails to compile with a type mismatch error where it used to work properly. Moving the attribute outside of the variant (`#[serde(deserialize_with="...")] Variant(Ty)`) seems to work on both and I believe should be equivalent and a valid workaround for now? Tested with serde_derive v1.0.29 (worked) and v1.0.57 (broken).

**Comment by dtolnay** [comment]
Sorry about the break! I released 1.0.58 with a fix.

**Comment by arcnmx** [comment]
Super responsive and quick as always, thanks a lot!
